submissions/management/commands/analyze_uploaded_file_sizes.py

```diff
@@ -0,0 +1,157 @@
+            default=today,
+            help=u"The most recent day to include in the report. Defaults to today."
+        )
+        parser.add_argument(
+            '--end_date',
+            type=parse_date,
+            default=thirty_days_ago,
+            help=(u"The longest-ago day to include in the report. Defaults to <today - 30 days>. "
+                  u"Cannot be set to before 2019-11-01.")
+        )
+
+    # pylint: disable=arguments-differ, unused-argument
+    def handle(self, start_date, end_date, *args, **options):
+        """
+        Analyze ORA file upload submissions and and print tab-limited report
+        """
+        print(u'Starting file upload submission report')
+        arg_echo_str = u'start_date = {}, end_date = {}'.format(
+            start_date.strftime(DATE_FORMAT),
+            end_date.strftime(DATE_FORMAT)
+        )
+        print(arg_echo_str)
+        end_date = self.validate_input_dates(start_date, end_date)
+        submission_data = self.load_data(start_date, end_date)
+        self.print_report(submission_data)
+
+    def validate_input_dates(self, start_date, end_date):
+        """
+        Checks the validity of start_date and end_date
+        Checks that the end date is not before LAST_DATE, and if it is, overrides it.
+        """
+        if start_date < end_date:
+            raise CommandError("End date must be before start date")
+        if datetime.date.today() < start_date:
+            print(u"Warning: start_date is in the future")
+        if end_date < LAST_DATE:
+            print(u'File size data is unavailable before 2019-11-01. Setting end_date to 2019-11-01')
+            return LAST_DATE
+        return end_date
+
+    def load_data(self, start_date, end_date):
+        """ Load the uploaded file data from the database """
+        # The model fields are datetimes, we need to make sure we
+        # include everything that was submitted in that day
+        end_datetime = self.beginning_of_day(end_date)
+        start_datetime = self.end_of_day(start_date)
+
+        return Submission.objects.filter(
+            created_at__range=(end_datetime, start_datetime),
+            student_item__item_type='openassessment',
+            answer__contains='files_sizes":['
+        ).order_by(
+            'student_item__course_id',
+            'student_item__student_id'
+        ).select_related('student_item').values_list(
+            'student_item__course_id',
+            'student_item__student_id',
+            'answer',
+        )
+
+    def beginning_of_day(self, dt):
+        """ Given a datetime, return a datetime that represents the first second of that day """
+        return datetime.datetime.combine(dt, datetime.time.min)
+
+    def end_of_day(self, dt):
+        """ Given a datetime, return a datetime that represents the last second of that day """
+        return datetime.datetime.combine(dt, datetime.time.max)
+
+    def print_report(self, submission_data):
```

Comment:
Could you update this function name, or docstring (or both) to indicate that this is where the "group by" is effectively done?